Lares/Assets/Player/Scripts/PlayerController.cs

```diff
@@ -0,0 +1,202 @@
+                if (!_isGroundedInternal)
+                    _movementForce *= _inAirMovementModifier;
+                else
+                    _movementForce /= _inAirMovementModifier;
+            }
+
+            get { return _isGroundedInternal; }
+        }
+
+        private bool _isGroundedInternal;
+
+        private int groundColliderCount;
+        private bool _canJump => (_isGrounded && _jumpCount == 0) || (!_isGrounded && _jumpCount == 1);
+        private int _jumpCount = 0;
+        //private bool _isOnIncline => OnIncline();
+        private bool _isMovementLocked;
+        private bool _isSprinting;
+
+        [Header("Raycast Settings")]
+        [SerializeField] float raycastDistance;
+
+        [Header("Move Settings")]
+        [SerializeField] private float _movementForce;
+        [SerializeField] private float _sprintSpeedMultiplier;
+        [SerializeField] private float _jumpForce;
+        [SerializeField] private float _rotationSpeed;
+        [SerializeField] private float _maxSpeed;
+        [SerializeField] private float _inAirMovementModifier;
+
+        [SerializeField] private Transform _cameraTransform;
+
+
+        [Header("New incline detection vars -- change this later")]
+        [SerializeField] private RaycastHit _inclineHitOut;
+        [SerializeField] private float _maxRaycastHitDistance = 1.0f;
+        [SerializeField] private float _maxInclineAngle = 60.0f;
+
+        // Start is called once before the first execution of Update after the MonoBehaviour is created
+        void Start()
+        {
+            _rigidbody = GetComponent<Rigidbody>();
+            _input = GetComponent<PlayerInput>();
+
+            _input.currentActionMap.FindAction("MovePlayer").performed += OnPlayerStartMoving;
+            _input.currentActionMap.FindAction("MovePlayer").canceled += OnPlayerStopMoving;
+            _input.currentActionMap.FindAction("Jump").performed += Jump;
+            //    //_input.currentActionMap.FindAction("Interact").performed += Interact;
+            _input.currentActionMap.FindAction("Sprint").performed += OnPlayerSprint;
+            _input.currentActionMap.FindAction("Sprint").canceled += OnPlayerSprint;
+        }
+
+        private void OnPlayerSprint(InputAction.CallbackContext context) { _isSprinting = !_isSprinting; }
+
+        private void OnPlayerStartMoving(InputAction.CallbackContext context)
+        {
+            Vector2 movementInput = context.ReadValue<Vector2>();
+
+            _moveDirection = new Vector3(movementInput.x, 0f, movementInput.y);
+            _moveCoroutine ??= StartCoroutine(Movement());
+        }
+
+        private void OnPlayerStopMoving(InputAction.CallbackContext context)
+        {
+            _moveDirection = Vector3.zero;
+        }
+
+        private IEnumerator Movement()
+        {
+            GameObject rotPoint = transform.GetChild(0).gameObject;
+            GameObject model = transform.GetChild(1).gameObject;
```

Comment:
This is very prone to error if you happen to ever change the order of children - could make these set through `SerializedField`s.